### PR TITLE
Test/jax environments

### LINKED ISSUE #524: [TEST] Jax environments
### What do you want to test?
Jax environments components

### Outline of test structure
* Unit tests
* Test components and hooks

### Definition of done
Passing checks, cover all hooks, edge cases considered

### Mandatory checklist before making a PR
* [ ] The success criteria laid down in “Definition of done” are met.
* [ ] Test code is documented - docstrings for methods and classes, static types for arguments.
* [ ] Documentation is updated - README, CONTRIBUTING, or other documentation.

### PR DESCRIPTION
**Pull request by mmorris44**
## What?
Test jax environment components.
## Why?
Robust testing of components before jax systems release.
## How?
- Tested `EnvironmentSpec`, `ExecutorEnvironmentLoop`, and `ParallelExecutorEnvironmentLoop`
## Extra
Closes #524

### CODE REVIEW

File: tests/jax/components/building/environments_test.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        assert environment.environment.num_agents == 10
+
+    def test_on_building_init_start(
+        self, test_environment_spec: EnvironmentSpec, test_builder: SystemBuilder
+    ) -> None:
+        """Test by manually calling the hook and checking the store."""
+        test_environment_spec.on_building_init_start(test_builder)
+
+        # Assert for type and extra spec
+        environment_spec = test_builder.store.environment_spec
```
Comment (RuanJohn) [inline tests/jax/components/building/environments_test.py:100]: Due to the env specs renaming PR this test currently has errors. Please update the names to match the new naming :)

File: tests/jax/components/building/environments_test.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    """Pytest fixture for executor environment loop"""
+
+    class ExecutorStatsWrapperClass:
+        def __init__(self, executor_environment_loop: Any) -> None:
+            self.executor_environment_loop = executor_environment_loop
+            self.wrapped = True
+
+    config = ExecutorEnvironmentLoopConfig(
+        should_update=False,
+        executor_stats_wrapper_class=ExecutorStatsWrapperClass,  # type: ignore
```
Comment (OmaymaMahjoub) [inline tests/jax/components/building/environments_test.py:63]: Why the executor_stats_wrapper_class is `ExecutorStatsWrapperClass` while it is mentioned as an object of `EnvironmentLoopStatisticsBase`?

File: tests/jax/components/building/environments_test.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    """Pytest fixture for executor environment loop"""
+    config = ExecutorEnvironmentLoopConfig(should_update=False)
+    return AbstractExecutorEnvironmentLoop(config=config)
+
+
+@pytest.fixture
+def test_parallel_executor_environment_loop() -> ParallelExecutorEnvironmentLoop:
+    """Pytest fixture for executor environment loop"""
+
+    class ExecutorStatsWrapperClass:
```
Comment (OmaymaMahjoub) [inline tests/jax/components/building/environments_test.py:56]: How about making this class inherits from `EnvironmentLoopStatisticsBase` so that you won't need `#type: ignore` in line 63

**Review by OmaymaMahjoub — DISMISSED** [review]
Looks good to me just a small optional change in line 56 if it is necessary!

File: tests/jax/components/building/environments_test.py (hunk 3 above)
Comment (mmorris44) [reply tests/jax/components/building/environments_test.py:56]: Thanks @OmaymaMahjoub. That's a good suggestion, but I think it might be quite involved, since I have to call to the __init__ of `EnvironmentLoopStatisticsBase`, which requires an actual environment loop. Happy to go for this if you'd like me to though?

Comment (OmaymaMahjoub) [reply tests/jax/components/building/environments_test.py:56]: Oh, I see yeah it makes sense that it needs to be a similar initiator if we do that, yeah as you like we can keep it like that, other than that I think it is a good PR thank you @mmorris44 !

**Review by OmaymaMahjoub — APPROVED** [review]

**Review by siddarthsingh1 — APPROVED** [review]
All looks good @mmorris44 . Everything seems to be covered :smile: .
